acme/challenge_test.go

```diff
@@ -457,7 +457,7 @@ func TestChallenge_Validate(t *testing.T) {
 						assert.Equal(t, "zap.internal", updch.Value)
 						assert.Equal(t, StatusPending, updch.Status)
 
-						err := NewError(ErrorConnectionType, "error doing http GET for url http://zap.internal/.well-known/acme-challenge/%s: force", ch.Token)
+						err := NewError(ErrorConnectionType, "error doing http GET for url http://zap.internal./.well-known/acme-challenge/%s: force", ch.Token)
```

Comment:
I think we shouldn't report the dot in the error messages, as it's different from what the user requested (most likely). Is it possible to keep the rooted name strictly internally?

Reply:
Agreed. Maybe could be an internal field on the Challenge?

Reply:
Fixed with 2b30ae50